x/protorev/keeper/epoch_hook.go

```diff
@@ -87,14 +91,33 @@ func (k Keeper) UpdatePools(ctx sdk.Context) error {
 // GetHighestLiquidityPools returns the highest liquidity pools for all base denoms
 func (k Keeper) GetHighestLiquidityPools(ctx sdk.Context, baseDenomPools map[string]map[string]LiquidityPoolStruct) error {
 	// Get all pools
-	pools, err := k.gammKeeper.GetPoolsAndPoke(ctx)
+	pools, err := k.poolmanagerKeeper.AllPools(ctx)
 	if err != nil {
 		return err
 	}
 
 	// Iterate through all pools and find valid matches
 	for _, pool := range pools {
-		coins := pool.GetTotalPoolLiquidity(ctx)
+		var coins sdk.Coins
+		poolType := pool.GetType()
+		if poolType == poolmanagertypes.Concentrated {
+			clPool, ok := pool.(cltypes.ConcentratedPoolExtension)
+			if !ok {
+				return fmt.Errorf("pool is not concentrated liquidity pool")
+			}
+
+			coins, err = k.concentratedLiquidityKeeper.GetTotalPoolLiquidity(ctx, clPool.GetId())
+			if err != nil {
+				return err
+			}
```

Comment:
What is the purpose of creating a clPool object if we already verify earlier that the poolType is a concentrated pool type. Seems like we can just use pool.GetId() in like 109 to get total liquidity and skip creating clPool, or am I missing something?

Reply:
it's so that we can use `ConcentratedPoolExtension` interface. Pool only has access to `poolmanagertypes.PoolI` which doesnot contain functions such as `GetTotalPoolLiquidity`

Reply:
where pool has a GetId() method as well and we use that instead of the clPool. But I think this seems fine since we check the type in the if statement above

```suggestion
			coins, err = k.concentratedLiquidityKeeper.GetTotalPoolLiquidity(ctx, pool.GetId())
			if err != nil {
				return err
			}
```

Reply:
yup that sounds good

Reply:
aah i see, this makes sense. But just making sure are we going to have X number of KV store for highest base_pair denom or just sticking with CL and gamm rn?

Reply:
Discussing this with David tomorrow, I'm kind of leaning towards not storing per module, and just figuring out the best way to store 1 pool per denom pair. Will let you know!

Reply:
hi @NotJeremyLiu  any update on this? ^ 

i'm planning on adding some test coverage to this PR and getting it R4R soon

Reply:
@stackman27 The takeaway was we will only store a single pool per denom pair, and compare across pool types. So the way it is currently done in this PR is fine. We decided that figuring out the best pools to use is a job for the hot_route method, and will put resources into that to make it more automated / intelligent to figure out what's best.

Reply:
lets merge #4966 into this PR and handle testing and other changes here